include/boost/url/any_allocator.hpp

```diff
@@ -0,0 +1,127 @@
+//
+// Created by alandefreitas on 12/01/2022.
+//
+
+#ifndef BOOST_URL_ANY_ALLOCATOR_HPP
+#define BOOST_URL_ANY_ALLOCATOR_HPP
+
+#include <cstddef>
+#include <type_traits>
+
+namespace boost {
+namespace urls {
+
+/** An allocator wrapper
+
+    This allocator erases the type of the underlying allocator
+
+ */
+template <class T>
+class any_allocator {
+  struct base;
+
+  base* p_ = nullptr;
+
+  template <class U>
+  friend class any_allocator;
+
+public:
+  /** The type any_allocator allocates
+   */
+  using value_type = char;
+
+  template<class U>
+  struct rebind
+  {
+    /** An any allocator for type U
+     */
+    using other = any_allocator<U>;
+  };
+
+  /** Determine with two allocator instances are always equal
+
+      This member is only optional since C++17
+   */
+  using is_always_equal = std::false_type;
+
+  /** Destructor
+   */
+  ~any_allocator();
+
+  /** Constructor
+  */
+  any_allocator() = default;
+
+  /** Constructor
+   */
+  any_allocator(
+      any_allocator const& other) noexcept;
+
+  /** Constructor
+   */
+  any_allocator(
+      any_allocator&& other) noexcept;
+
+  /** Constructor
+   */
+  template <class U>
+  any_allocator(
+      any_allocator<U> const& other) noexcept;
+
+  /** Constructor
+   */
+  template <class U>
+  any_allocator(
+      any_allocator<U> && other) noexcept;
+
+  /** Copy assignment
+   */
+  any_allocator&
+  operator=(
+      any_allocator const& other) noexcept;
+
+  /** Constructor
+   */
+  template<class Allocator>
+  explicit
+  any_allocator(Allocator const& a);
+
+  /** Allocate a string with the underlying allocator
```

Comment:
We are removing these comments but it is wrong anyway

Reply:
Yes. That was from before.